validator/valnode/valnode.go

```diff
@@ -0,0 +1,126 @@
+	Arbitrator server_arb.ArbitratorSpawnerConfig `koanf:"arbitrator" reload:"hot"`
+	Jit        server_jit.JitSpawnerConfig        `koanf:"jit" reload:"hot"`
+	Wasm       WasmConfig                         `koanf:"wasm"`
+}
+
+type ValidationConfigFetcher func() *Config
+
+var DefaultValidationConfig = Config{
+	Jit:        server_jit.DefaultJitSpawnerConfig,
+	ApiAuth:    true,
+	ApiPublic:  false,
+	Arbitrator: server_arb.DefaultArbitratorSpawnerConfig,
+	Wasm:       DefaultWasmConfig,
+}
+
+var TestValidationConfig = Config{
+	Jit:        server_jit.DefaultJitSpawnerConfig,
+	ApiAuth:    false,
+	ApiPublic:  true,
+	Arbitrator: server_arb.DefaultArbitratorSpawnerConfig,
+	Wasm:       DefaultWasmConfig,
+}
+
+func ValidationConfigAddOptions(prefix string, f *flag.FlagSet) {
+	f.Bool(prefix+".use-jit", DefaultValidationConfig.UseJit, "use jit for validation")
+	f.Bool(prefix+".api-auth", DefaultValidationConfig.ApiAuth, "validate is a public API")
+	f.Bool(prefix+".api-public", DefaultValidationConfig.ApiPublic, "validate is an authenticated API")
+	server_arb.ArbitratorSpawnerConfigAddOptions(prefix+".arbitrator", f)
+	server_jit.JitSpawnerConfigAddOptions(prefix+".jit", f)
+	WasmConfigAddOptions(prefix+".wasm", f)
+}
+
+type ValidationNode struct {
+	config     ValidationConfigFetcher
+	arbSpawner *server_arb.ArbitratorSpawner
+	jitSpawner *server_jit.JitSpawner
+}
+
+func CreateValidationNode(configFetcher ValidationConfigFetcher, stack *node.Node, fatalErrChan chan error) (*ValidationNode, error) {
+	config := configFetcher()
+	locator, err := server_common.NewMachineLocator(config.Wasm.RootPath)
+	if err != nil {
+		return nil, err
+	}
+	arbConfigFetcher := func() *server_arb.ArbitratorSpawnerConfig {
+		return &configFetcher().Arbitrator
+	}
+	arbSpawner, err := server_arb.NewArbitratorSpawner(locator, arbConfigFetcher)
+	if err != nil {
+		return nil, err
+	}
+	node.DefaultAuthModules = []string{server_api.Namespace}
+	var serverAPI *server_api.ExecServerAPI
+	var jitSpawner *server_jit.JitSpawner
+	if config.UseJit {
+		jitConfigFetcher := func() *server_jit.JitSpawnerConfig { return &configFetcher().Jit }
+		var err error
+		jitSpawner, err = server_jit.NewJitSpawner(locator, jitConfigFetcher, fatalErrChan)
+		if err != nil {
+			return nil, err
+		}
+		serverAPI = server_api.NewExecutionServerAPI(jitSpawner, arbSpawner)
+	} else {
+		serverAPI = server_api.NewExecutionServerAPI(arbSpawner, arbSpawner)
+	}
+	valAPIs := []rpc.API{{
+		Namespace:     server_api.Namespace,
+		Version:       "1.0",
+		Service:       serverAPI,
+		Public:        config.ApiPublic,
+		Authenticated: config.ApiAuth,
+	}}
+	stack.RegisterAPIs(valAPIs)
+
+	return &ValidationNode{configFetcher, arbSpawner, jitSpawner}, nil
+}
+
+func (v *ValidationNode) Start(ctx context.Context) error {
+	if err := v.arbSpawner.Start(ctx); err != nil {
+		return err
+	}
+	if v.jitSpawner == nil {
+		return nil
+	}
+	if err := v.jitSpawner.Start(ctx); err != nil {
+		return err
```

Comment:
It might be worth just putting this in an `if v.jitSpawner != nil` instead of returning early. My thinking is that we could add some other thing we need to start to this function, and we might not run it by accident if the jit validator is disabled because we returned early.